### ISSUE TITLE
Add a rule to require job's `timeout-minutes`

### ISSUE THREAD

**Issue by suzuki-shunsuke** [suzuki-shunsuke/ghalint#485]
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes

https://exercism.org/docs/building/github/gha-best-practices#h-set-timeouts-for-workflows

> By default, GitHub Actions kills workflows after 6 hours if they have not finished by then. Many workflows don't need nearly as much time to finish, but sometimes unexpected errors occur or a job hangs until the workflow run is killed 6 hours after starting it. Therefore it's recommended to [specify a shorter timeout](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes).
> 
> The ideal timeout depends on the individual workflow but 30 minutes is typically more than enough for the workflows used in Exercism repos.
> 
> This has the following advantages:
> 
> PRs won't be pending CI for half the day, issues can be caught early or workflow runs can be restarted.
> The number of overall parallel builds is limited, hanging jobs will not cause issues for other PRs if they are cancelled early.

**Comment by suzuki-shunsuke** [comment]
`jobs.<job_id>.timeout-minutes`

```yaml
jobs:
  foo:
    timeout-minutes: 30
```
